native/native.tests/testData/codegen/cinterop/kt66169.kt

```diff
@@ -0,0 +1,44 @@
+// TARGET_BACKEND: NATIVE
+// NATIVE_STANDALONE
```

Comment:
If I don't make the test standalone, it gets ignored when I attempt to run it locally. What dependency am I missing here?

Reply:
Seems working fine for me. What do you mean by "gets ignored"? How do I reproduce that?

Reply:
If I remove `// NATIVE_STANDALONE` and run the test with confgiuration `:native:native.tests:codegenBoxTest --tests "org.jetbrains.kotlin.konan.test.blackbox.NativeCodegenLocalTestGenerated$Cinterop.testKt66169"`, it is ignored.

Reply:
I can't confirm this. Could you please describe step-by-step what you do and what you observe precisely?
In particular, do you run it from the terminal or from an IDE?

Reply:
I ran the tests from the IDE. Below you can see the step-by-step reproducer and what I observe.


1.) delete `// NATIVE_STANDALONE`
![step1](https://github.com/JetBrains/kotlin/assets/65320245/ac021a2f-01e3-4bca-9ecc-880b179e4d84)

2.) run test from the IDE with selecting the `test` task
![step2](https://github.com/JetBrains/kotlin/assets/65320245/76435abd-29ae-4e88-a739-f628009e27fb)

3.) inspect the `Run` window, that says the test is ignored
![step3](https://github.com/JetBrains/kotlin/assets/65320245/df87683e-30f4-46f5-a7d2-37a628ae5b53)

If I add the  `// NATIVE_STANDALONE` line and repeat step 2, the IDE says the test passed. Also it takes longer to run. I guess I'm either missing some dependency or have a misconfigured build system.
![step4](https://github.com/JetBrains/kotlin/assets/65320245/b339ab12-a4f5-4fc6-982f-2dc147f36135)

Reply:
> I ran the tests from the IDE. Below you can see the step-by-step reproducer and what I observe.

I do the same, and the test is still marked as passed on my side. What version of IDEA do you use?
Let's try to debug this. If the test is marked as ignored, it probably calls `org.junit.jupiter.api.Assumptions.throwAssumptionFailed` at some point. Could you please set a breakpoint there, run the test without `NATIVE_STANDALONE` and show where it stops? Alternatively, a breakpoint can be set for the `org.opentest4j.TestAbortedException` exception, I guess.

"NATIVE_STANDALONE" is basically a hack needed for specific tests that don't work with test grouping for various reasons. That doesn't seem the case here, so it is desirable to remove this line.

More importantly, if our test infrastructure just marks some tests as ignored without a reason, I would really like to know that. So your assistance would be truly helpful.

Reply:
I use IDEA 2024.1.1 (Community Edition) and I don't hit `throwAssumptionFailed`. Apparently the test doesn't even start...

I debugged it further and in `modulesToKlib` I eventually hit `muteCInteropTestIfNecessary` and hit `Assumptions.abort<Nothing>("C-interop tests with Objective-C are not supported at non-Apple targets, def file: $defFile")`. The test suite wants to compile `kt55938.kt` too, which has `language = Objective-C` specified in it's header.

The problem is in `withTestExecutable()`.
```kotlin
val testCompilation = when (testCase.kind) {
    TestKind.STANDALONE, TestKind.STANDALONE_NO_TR, TestKind.STANDALONE_LLDB -> {
        // Create a separate compilation for each standalone test case.
        cachedCompilations.computeIfAbsent(
            TestCompilationCacheKey.Standalone(testCaseId)
        ) {
            compilationFactory.testCasesToExecutable(listOf(testCase), settings) // <-- we compile only the single test we need
        }
    }
    TestKind.REGULAR -> {
        // Group regular test cases by compiler arguments.
        val testRunnerType = testCase.extras<WithTestRunnerExtras>().runnerType
        cachedCompilations.computeIfAbsent(
            TestCompilationCacheKey.Grouped(
                testCaseGroupId = testCaseGroupId,
                freeCompilerArgs = testCase.freeCompilerArgs,
                sharedModules = testCase.sharedModules,
                runnerType = testRunnerType
            )
        ) {
            val testCases = testCaseGroup.getRegularOnly(testCase.freeCompilerArgs, testCase.sharedModules, testRunnerType)
            assertTrue(testCases.isNotEmpty())
            compilationFactory.testCasesToExecutable(testCases, settings) // <-- we compile the whole group
        }
    }
}
```

To experiment with it, I removed the `when` expression and changed `testCompilation` like this:
```kotlin
// Create a separate compilation for each standalone test case.
val testCompilation = cachedCompilations.computeIfAbsent(
    TestCompilationCacheKey.Standalone(testCaseId)
) {
    compilationFactory.testCasesToExecutable(listOf(testCase), settings)
}
```
As a result I get 63 out of 82 tests passed with the remaining 19 Objective-C tests being ignored.

I'm not sure why we want to compile the whole group by default. @ddolovov can you please help to figure this out?

Reply:
Compiling tests in groups is an optimization that gives a significant performance boost at CI as compared to compiling each test in isolation from others. Only in rare specific cases tests are marked as "standalone" (which is an equivalent of "compile me in isolation from others").

Reply:
I see, but now because of the grouping if one of the tests is ignored, the whole group will be ignored. Since the group contains test files that can only be run on apple devices, locally it's only possible to run the tests properly on apple devices only.

Reply:
Nice finding!

As Dmitriy said, the whole grouping thing is an optimization.

The fact that a single test makes the whole group ignored is definitely a bug in the test infrastructure. I've submitted it here: https://youtrack.jetbrains.com/issue/KT-68416/Native-when-using-test-grouping-a-whole-group-gets-ignored-on-non-Mac-hosts-if-it-has-Objective-C-tests

Meanwhile, I propose the following: please either remove `// NATIVE_STANDALONE` from the test or add a comment mentioning the issue.

Reply:
I removed `// NATIVE_STANDALONE`.